test/PlaidML/CppEdsl.DotF16.mlir

```diff
@@ -0,0 +1,42 @@
+// RUN: %python_executable %imex_runner -i %s --pass-pipeline-file=%p/linalg-to-cpu.pp \
+//                                            --runner mlir-cpu-runner -e main \
+//                                            --shared-libs=%mlir_runner_utils \
+//                                            --entry-point-result=void | FileCheck %s
+#map0 = affine_map<(d0, d1, d2) -> (d0, d2)>
+#map1 = affine_map<(d0, d1, d2) -> (d2, d1)>
+#map2 = affine_map<(d0, d1, d2) -> (d0, d1)>
+module @dot_f16 {
+  func.func @test(%arg0: tensor<8x16xf16>, %arg1: tensor<16x32xf16>) -> tensor<8x32xf16> {
+    %cst = arith.constant 0.000000e+00 : f16
+    %0 = tensor.empty() : tensor<8x32xf16>
+    %1 = linalg.fill ins(%cst : f16) outs(%0 : tensor<8x32xf16>) -> tensor<8x32xf16>
+    %2 = linalg.generic {indexing_maps = [#map0, #map1, #map2], iterator_types = ["parallel", "parallel", "reduction"]} ins(%arg0, %arg1 : tensor<8x16xf16>, tensor<16x32xf16>) outs(%1 : tensor<8x32xf16>) attrs =  {iterator_ranges = [8, 32, 16]} {
+    ^bb0(%arg2: f16, %arg3: f16, %arg4: f16):
+      %3 = arith.mulf %arg2, %arg3 : f16
+      %4 = arith.addf %arg4, %3 : f16
+      linalg.yield %4 : f16
+    } -> tensor<8x32xf16>
+    return %2 : tensor<8x32xf16>
+  }
+
+  func.func @main() {
+    %0 = arith.constant dense<[[1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0, 11.0, 12.0, 13.0, 14.0, 15.0, 16.0],
+                               [1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0, 11.0, 12.0, 13.0, 14.0, 15.0, 16.0],
+                               [1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0, 11.0, 12.0, 13.0, 14.0, 15.0, 16.0],
+                               [1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0, 11.0, 12.0, 13.0, 14.0, 15.0, 16.0],
+                               [1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0, 11.0, 12.0, 13.0, 14.0, 15.0, 16.0],
+                               [1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0, 11.0, 12.0, 13.0, 14.0, 15.0, 16.0],
+                               [1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0, 11.0, 12.0, 13.0, 14.0, 15.0, 16.0],
+                               [1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0, 11.0, 12.0, 13.0, 14.0, 15.0, 16.0]
+                              ]> : tensor<8x16xf16>
+    %1 = arith.constant dense<1.0> : tensor<16x32xf16>
+    %2 = call @test(%0, %1) : (tensor<8x16xf16>, tensor<16x32xf16>) -> tensor<8x32xf16>
+    %unranked = tensor.cast %2 : tensor<8x32xf16> to tensor<*xf16>
+    call @printMemrefF16(%unranked) : (tensor<*xf16>) -> ()
+    //      CHECK: Unranked Memref base@ = {{(0x)?[-9a-f]*}}
+    // CHECK-NEXT: [136, 136, 136, 136, 136, 136, 136, 136, 136, 136, 136, 136, 136, 136, 136, 136, 136, 136, 136, 136, 136, 136, 136, 136, 136, 136, 136, 136, 136, 136, 136, 136]
+    return
+  }
+
+  func.func private @printMemrefF16(%ptr : tensor<*xf16>)
```

Comment:
where is this print function in upstream? I dont see it

Reply:
It seems not available in upstream. Will focus on F32 first and left FP16 support for the future